File: backend/API/tg_bots/Model/Category.py
from pydantic import BaseModel
from typing import List, Optional, Union

from Model.MenuItem import MenuItem
from Model.Dish import Dish
from Model.Drink import Drink


class Category(BaseModel):
    id: Optional[int]
    menu_id: int
    name: str
    menu_items: Optional[List[Union[Dish, Drink]]]

    def __init__(self,
                 menu_id: int,
                 name: str,
                 id: Optional[int] = None,
                 menu_items: Optional[List[Union[Dish, Drink]]] = None):
        super().__init__(
            id=id,
            menu_id=menu_id,
            name=name,
            menu_items=menu_items
        )

File: backend/Domain/Models/Category.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Domain.DataBaseAccess;
using Microsoft.EntityFrameworkCore;

namespace Domain.Models;

public class Category : ITableDataBase, ITakeRelatedData, IDeleteRelatedData
{
    public long? Id { get; set; }
    [JsonPropertyName("menu_id")] public long MenuId { get; init; }
    public string Name { get; init; }
    [JsonPropertyName("menu_items")] [NotMapped] public List<MenuItem>? MenuItems { get; set; }
    
    public Category() {}
    public Category(long? id, long menuId, string name, List<MenuItem>? menuItems)
    {
        Id = id;
        MenuId = menuId;
        Name = name;
        MenuItems = menuItems;
    }

    public async Task TakeRelatedData(ApplicationDbContext context)
    {
        MenuItems = new List<MenuItem>();

        foreach (var dish in await context.Dishes
                     .Where(d => d.CategoryId == Id).ToListAsync())
        {
            await dish.TakeRelatedData(context);
            MenuItems.Add(dish);
        }

        foreach (var drink in await context.Drinks
                     .Where(d => d.CategoryId == Id).ToListAsync())
        {
            await drink.TakeRelatedData(context);
            MenuItems.Add(drink);
        }
    }

    public void DeleteRelatedData(ApplicationDbContext context)
    {
        foreach (var dish in context.Dishes
                     .Where(d => d.CategoryId == Id))
        {
            context.Dishes.Remove(dish);
            context.SaveChanges();
            dish.DeleteRelatedData(context);
        }
        foreach (var drink in context.Drinks
                     .Where(d => d.CategoryId == Id))
        {
            context.Drinks.Remove(drink);
            context.SaveChanges();
            drink.DeleteRelatedData(context);
        }
    }
}